File: Python/7py 2/mygcd.py
"""
|--------------------------------------------------------------------------
| Greatest common divisor
|--------------------------------------------------------------------------
|
"""

def mygcd(x, y):
   return mygcd(y, x % y) if y else x


def mygcd(x,y):
   while y:
      x,y=y,x%y
   return x


from math import gcd as mygcd


print(mygcd(30,12),6)
print(mygcd(8,9),1)
print(mygcd(1,1),1)

File: JavaScript/7js 2/mygcd.js
/*
|--------------------------------------------------------------------------
| Greatest common divisor
|--------------------------------------------------------------------------
|
*/

function mygcd(x, y) {
   return y ? mygcd(y, x % y) : x;
}


function mygcd(x, y) {
   while (y) {
      [x, y] = [y, x % y];
   }
   return x;
}


function mygcd(x, y) {
   for (let i = x; i > 0; i -= 1) {
      if (x % i === 0 && y % i === 0) {
         return i;
      }
   }
}


console.log(mygcd(30, 12), 6);
console.log(mygcd(8, 9), 1);
console.log(mygcd(1, 1), 1);